api/v1/service/dashboard/dashboard.go

```diff
@@ -569,59 +569,97 @@ where t.chain_id = ?
-    from s
-        join pair p on p.id = s.pair_id
-        join s_7d  on p.id = s_7d.pair_id
-        left join tokens t on p.chain_id = t.chain_id and (p.asset0 = t.address or p.asset1 = t.address)
+WITH
+recent_48h AS (
+    SELECT
+        ps.pair_id,
+        SUM(CASE
+            WHEN ps.timestamp >= extract(epoch from now() - interval '1 day')
+            THEN ps.volume0_in_price ELSE 0 END) AS vol0_24h,
+        SUM(CASE
+            WHEN ps.timestamp >= extract(epoch from now() - interval '1 day')
+            THEN ps.volume1_in_price ELSE 0 END) AS vol1_24h,
+        SUM(CASE
+            WHEN ps.timestamp < extract(epoch from now() - interval '1 day')
+            THEN ps.volume0_in_price ELSE 0 END) AS vol0_prev,
+        SUM(CASE
+            WHEN ps.timestamp < extract(epoch from now() - interval '1 day')
+            THEN ps.volume1_in_price ELSE 0 END) AS vol1_prev,
+        SUM(CASE
+            WHEN ps.timestamp >= extract(epoch from now() - interval '1 day')
+            THEN ps.commission0_in_price ELSE 0 END) AS com0_24h,
+        SUM(CASE
+            WHEN ps.timestamp >= extract(epoch from now() - interval '1 day')
+            THEN ps.commission1_in_price ELSE 0 END) AS com1_24h
+    FROM pair_stats_recent ps
+    GROUP BY ps.pair_id
+),
+recent_14d AS (
+    SELECT
+        ps.pair_id,
+        SUM(CASE
+            WHEN ps.timestamp >= extract(epoch from now() - interval '7 day')
+            THEN ps.volume0_in_price ELSE 0 END) AS vol0_7d,
+        SUM(CASE
+            WHEN ps.timestamp >= extract(epoch from now() - interval '7 day')
+            THEN ps.volume1_in_price ELSE 0 END) AS vol1_7d,
+        SUM(CASE
+            WHEN ps.timestamp < extract(epoch from now() - interval '7 day')
+             AND ps.timestamp >= extract(epoch from now() - interval '14 day')
+            THEN ps.volume0_in_price ELSE 0 END) AS vol0_prev,
+        SUM(CASE
+            WHEN ps.timestamp < extract(epoch from now() - interval '7 day')
+             AND ps.timestamp >= extract(epoch from now() - interval '14 day')
+            THEN ps.volume1_in_price ELSE 0 END) AS vol1_prev
+    FROM pair_stats_30m ps
+    WHERE ps.timestamp >= extract(epoch from now() - interval '14 day')
+    GROUP BY ps.pair_id
+),
+latest_pair_tvl AS (
+    SELECT DISTINCT ON (ps.pair_id)
+        ps.pair_id,
+        ps.liquidity0_in_price,
+        ps.liquidity1_in_price
+    FROM pair_stats_30m ps
+    ORDER BY ps.pair_id, ps.timestamp DESC
+),
+pair_tvl_24h_before AS (
+    SELECT DISTINCT ON (ps.pair_id)
+        ps.pair_id,
+        ps.liquidity0_in_price,
+        ps.liquidity1_in_price
+    FROM pair_stats_30m ps
+    WHERE ps.timestamp < extract(epoch from now() - interval '1 day')
+    ORDER BY ps.pair_id, ps.timestamp DESC
+)
+SELECT
+    address,
+    COALESCE(volume_24h, 0) AS volume,
+    COALESCE((volume_24h - volume_24h_before) / GREATEST(volume_24h_before, 1), 0) AS volume_change,
+    COALESCE(volume_7d + volume_24h, 0) AS volume_week,
```

Comment:
nit; any way to deduct the first day of `volume_7d` for deriving more exact `volume_week`?

Reply:
Plus, how about changing to `COALESCE(volume_7d, 0) + COALESCE(volume_24h, 0)` for preventing 0 return in case of (non-null `volume_7d` + null `volume_24h`) = null?

Reply:
removed adding volume_24h, I guess it'd be better to be a bit inaccurate rather than to calculate and add 30 mins from recent table.